Services/Music_Portal.Services.Services/ArtistService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Music_Portal.Domain.Core;
+using Music_Portal.Domain.Interfaces;
+using Music_Portal.Services.Interfaces;
+
+namespace Music_Portal.Services.Services
+{
+    public class ArtistService : IArtistService
+    {
+        private readonly ILastFmService _lastFmService;
+        private readonly IMapper _mapper;
+        private readonly IArtistRepository _repository;
+
+        public ArtistService(ILastFmService lastFmService, IMapper mapper, IArtistRepository repository)
+        {
+            _lastFmService = lastFmService;
+            _mapper = mapper;
+            _repository = repository;
+        }
+
+        public async Task<IEnumerable<Artist>> GetTopArtists()
+        {
+            if (_repository.GetArtistsEnumerable() == null)
```

Comment:
You have 2 requests to the database, but you can do it only once, just move the call to GetArtistsEnumerable to the separate line with variable and use variable where you want your data